frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx

```diff
@@ -1338,14 +1342,42 @@ class DatasetSettings extends React.PureComponent<DatasetSettingsProps, State> {
       ),
       onOk: async () => {
         try {
-          const { flycam } = Store.getState();
+          const { flycam, temporaryConfiguration } = Store.getState();
           const position = V3.floor(getPosition(flycam));
           const zoom = flycam.zoomStep;
+          const { activeMappingByLayer } = temporaryConfiguration;
           const completeDatasetConfiguration = Object.assign({}, datasetConfiguration, {
             position,
             zoom,
           });
-          await updateDatasetDefaultConfiguration(dataset, completeDatasetConfiguration);
+          const updatedLayers = {
+            ...completeDatasetConfiguration.layers,
+          } as DatasetConfiguration["layers"];
+          Object.keys(activeMappingByLayer).forEach((layerName) => {
+            const mappingInfo = activeMappingByLayer[layerName];
+            if (
+              mappingInfo.mappingStatus === MappingStatusEnum.ENABLED &&
+              mappingInfo.mappingName != null
+            ) {
+              updatedLayers[layerName] = {
+                ...updatedLayers[layerName],
+                mapping: {
+                  name: mappingInfo.mappingName,
+                  type: mappingInfo.mappingType,
+                },
+              };
+            } else {
+              updatedLayers[layerName] = {
```

Comment:
This is the default Store value anyway, right? No need to reassign it.

Reply:
> This is the default Store value anyway, right?

I'd say no: I modifed the type of the DatasetConfiguration so that layers may have the mapping setting (which is null by default). But in case an admin sets a mapping for layer A, goes into viewmode, disables the mapping for layer A and than saves these changes, `datasetConfiguration.layers` should still contain a mapping for layer A as it was previously served this way by the backend.

=> I just tested this scenario with the debugger and was able to confirm that this case is necessery. Else one cannot deactivate a mapping previously activated by useing the "Save View Configuration as Default" feature.